db/db_impl_open.cc

```diff
@@ -892,7 +892,7 @@ Status DBImpl::WriteLevel0TableForRecovery(int job_id, ColumnFamilyData* cfd,
           std::unique_ptr<InternalIterator>(mem->NewRangeTombstoneIterator(ro)),
           &meta, cfd->internal_comparator(),
           cfd->int_tbl_prop_collector_factories(), cfd->GetID(), cfd->GetName(),
-          snapshot_seqs, earliest_write_conflict_snapshot,
+          snapshot_seqs, earliest_write_conflict_snapshot, nullptr,
```

Comment:
why no snapshot_checker is passed here?

Reply:
I'm wanting to avoid adding yet another option to db_options, and we don't really need snapshot_checker before WritePreparedTxnDB init, which is after DBImpl open.

Reply:
Then if we use SetSnapshotChecker to set it, does anybody uses this changed API of BuildTable to set the snapshot checker? If not can we remove it?